Modules/Common/src/ReferenceComparatorCheck.cxx

```diff
@@ -129,6 +133,32 @@ static Quality compare(TCanvas* canvas, ObjectComparatorInterface* comparator, s
   return comparator->compare(plots.first, plots.second, message);
 }
 
+Quality ReferenceComparatorCheck::getSinglePlotQuality(std::shared_ptr<MonitorObject> mo, std::string& message)
+{
+  // retrieve the reference plot and compare
+  auto* th1 = dynamic_cast<TH1*>(mo->getObject());
+  auto referenceRun = std::stoi(mCustomParameters.atOptional("referenceRun").value_or("0"));
+
+  // get path of mo and ref (we have to remove the provenance)
+  string path = mo->getPath();
+  size_t pos = path.find('/');
+  if (pos != std::string::npos) {
+    path = path.substr(pos + 1);
+  }
+
+  auto referencePlot = retrieveReference(path, referenceRun, mActivity);
+  if (!referencePlot) {
+    message = "The reference plot is empty";
```

Comment:
this message might be misleading. It could be understood as "the reference plot is there, but there are no entries". Please rephrase to convey the meaning "could not retrieve the reference Monitor Object".